### ISSUE TITLE
adding local jar dependencies

### ISSUE THREAD

**Issue by amiracam** [jruby-gradle/jruby-gradle-plugin#313]
what if we wanted to include a "local" jar ? i.e. jars we had internally created or from external parties which do not provide Maven coordinates but rather just a download ?

I noticed that JRubyPrepare.groovy has the following code snippet:

>  @Optional
>     /** Adds dependencies from the given configuration to be prepared
>      *
>      * @param f A file, directory, configuration or list of gems
>      */
>     void dependencies(Object f) {
>         dependencies.add(f)
>     }

so I tried this:

dependencies {
    '/Users/charles/IdeaProjects/JrubyGradleTest5/local_jars/ojdbc6-11.2.0.3.jar'
}

figured an absolute path would make for a more direct test

Running JRubyPrepare task however did not capture above specified jar.

Is there perhaps a way of specifying "local-ness" via a jrubyJar entry 
e.g. 
jrubyJar 'file:///Users/charles/IdeaProjects/JrubyGradleTest5/local_jars/ojdbc6-11.2.0.3.jar'

thanks

**Comment by ysb33r** [comment]
You can create an additional configuration, add the JARs to that using `files`, then use an additional `from` in the jrubyJar config.


Sent from my Samsung device

-------- Original message --------

**Comment by amiracam** [comment]
well what I did was to add the apply the java plugin , setup a repository with flatDir and then setup up the jrubyJar task

so:

apply plugin: 'java'
repositories {
    flatDir {
        dirs 'local_jars'
    }
}

dependencies {
    compile name: 'ojdbc6-11.2.0.3'
}

jrubyJar {
    /* We want to use this Ruby script as our start point when the jar executes
     */
    jrubyVersion '9.1.13.0'
    initScript "${projectDir}/src/entrypoint.rb"
    from configurations.compile
}

that worked but placed the jar at the top of the jar , not sure if this makes it accessible to the code to i.e. to the jar-bootstrap.rb which so far is the only code I am able to include

**Comment by ysb33r** [comment]
I meant do something like this:

```groovy
configuration {
  localJars
}

dependencies {
  localJars ':ojdbc6:11.2.0.3'
}

jrubyJar {
  from configurations.localJars, {
    into 'mySubFolderInJRubyJar'
  }
}
```

**Comment by amiracam** [comment]
yes, I actually had tried something similar :) which did not work, I say this as I'm eating my breakfast yogurt watching a video on Gradle from safari online :) , thanks for the help, I'll play around, btw, I would have thought that you all would have tried to provide similar features to Warbler i.e. order to become their replacement

**Comment by ysb33r** [comment]
That is generally the idea. @rtyler has more experience in the Warbler area. 

Of course, we only add feastures as use-cases are discovered and we have time.
